library/VTKExtensions/Readers/vtkF3DGenericImporter.cxx

```diff
@@ -116,138 +222,110 @@ bool vtkF3DGenericImporter::GetTemporalInformation(
-  vtkImageData* image = vtkImageData::SafeDownCast(this->PostPro->GetOutput(2));
-
-  // Configure volume mapper
-  this->VolumeMapper->SetInputConnection(this->PostPro->GetOutputPort(2));
-  this->VolumeMapper->SetRequestedRenderModeToGPU();
-
-  // Configure polydata mapper
-  this->PolyDataMapper->InterpolateScalarsBeforeMappingOn();
-  this->PolyDataMapper->SetInputConnection(this->PostPro->GetOutputPort(0));
-
-  // Configure Point Gaussian mapper
-  this->PointGaussianMapper->SetInputConnection(this->PostPro->GetOutputPort(1));
-  this->PointGaussianMapper->EmissiveOff();
-  this->PointGaussianMapper->SetSplatShaderCode(
-    "//VTK::Color::Impl\n"
-    "float dist = dot(offsetVCVSOutput.xy, offsetVCVSOutput.xy);\n"
-    "if (dist > 1.0) {\n"
-    "  discard;\n"
-    "} else {\n"
-    "  float scale = (1.0 - dist);\n"
-    "  ambientColor *= scale;\n"
-    "  diffuseColor *= scale;\n"
-    "}\n");
-
-  vtkDataSet* dataSet = vtkImageData::SafeDownCast(this->PostPro->GetInput())
-    ? vtkDataSet::SafeDownCast(image)
-    : vtkDataSet::SafeDownCast(surface);
-
-  this->PointDataForColoring = vtkDataSetAttributes::SafeDownCast(dataSet->GetPointData());
-  this->CellDataForColoring = vtkDataSetAttributes::SafeDownCast(dataSet->GetCellData());
-
-  // configure props
-  this->VolumeProp->SetMapper(this->VolumeMapper);
-
-  this->GeometryActor->SetMapper(this->PolyDataMapper);
-  this->GeometryActor->GetProperty()->SetInterpolationToPBR();
-
-  this->GeometryActor->GetProperty()->SetColor(this->SurfaceColor);
-  this->GeometryActor->GetProperty()->SetOpacity(this->Opacity);
-  this->GeometryActor->GetProperty()->SetRoughness(this->Roughness);
-  this->GeometryActor->GetProperty()->SetMetallic(this->Metallic);
-  this->GeometryActor->GetProperty()->SetLineWidth(this->LineWidth);
-  this->GeometryActor->GetProperty()->SetPointSize(this->PointSize);
-
-  this->PointSpritesActor->SetMapper(this->PointGaussianMapper);
-  this->PointSpritesActor->GetProperty()->SetColor(this->SurfaceColor);
-  this->PointSpritesActor->GetProperty()->SetOpacity(this->Opacity);
-
-  // Textures
-  auto colorTex = this->GetTexture(this->TextureBaseColor, true);
-  this->GeometryActor->GetProperty()->SetBaseColorTexture(colorTex);
-  this->GeometryActor->GetProperty()->SetORMTexture(this->GetTexture(this->TextureMaterial));
-  this->GeometryActor->GetProperty()->SetEmissiveTexture(
-    this->GetTexture(this->TextureEmissive, true));
-  this->GeometryActor->GetProperty()->SetEmissiveFactor(this->EmissiveFactor);
-  this->GeometryActor->GetProperty()->SetNormalTexture(this->GetTexture(this->TextureNormal));
-  this->GeometryActor->GetProperty()->SetNormalScale(this->NormalScale);
-
-  // If the input texture is RGBA, flag the actor as translucent
-  if (colorTex && colorTex->GetImageDataInput(0)->GetNumberOfScalarComponents() == 4)
+  // Update each reader
+  for (size_t readerIndex = 0; readerIndex < this->Internals->Readers.size(); readerIndex++)
```

Comment:
Is it streamed or the actors are visible only when everything is loaded?

Reply:
Not streamed. Loading all is needed however if a reader has already read the file, it will not read again.